### ISSUE TITLE
Signature mismatch issues when using @abstractmethod

### ISSUE THREAD

**Issue by Kangaroux** [python/mypy#8966]
# **TL;DR**

1. [comment](https://github.com/python/mypy/issues/8966#issuecomment-640654472) - There is no error thrown when a subclass implements an abstract method but uses too many/few arguments **and** the subclass method does not include any type annotations in its signature
2. [comment](https://github.com/python/mypy/issues/8966#issuecomment-640663342) - When a subclass implements an abstract method the types from the method signature are not inherited properly. Omitting a type from the signature defaults to `Any` instead of the type from the abstract method.

---

1. Abstract class declares an abstract method with args and/or a return type
2. Subclass implements the abstract method, but does not have args or a return type
3. No error is thrown

```python
from abc import ABC, abstractmethod

class Interface(ABC):
    @abstractmethod
    def abstract_method(self, a: int) -> str:
        raise NotImplementedError

class Concrete(Interface):
    def abstract_method(self):  # Signature mismatch
        return 123
```

This works as expected if it's the other way around (empty sig on Interface)

Python 3.7.5, mypy 0.780

**Comment by Kangaroux** [comment]
I found another issue that is likely more the root cause.

If an abstract method is incorrectly implemented **and does not have any type annotations** then no error is thrown.

```python
class Concrete(Interface):
    def abstract_method(self, a):  # No error
        return 123

class Concrete(Interface):
    def abstract_method(self, a, b, c, d, e):  # No error
        return 123

class Concrete(Interface):
    def abstract_method(self, a: int):  # Throws an error
        return 123
```

**Comment by Kangaroux** [comment]
The lack of type annotations also causes the implemented method to not inherit the types of the abstract method:

```python
from abc import ABC, abstractmethod

class Interface(ABC):
    @abstractmethod
    def abstract_method(self, a: int) -> str:
        raise NotImplementedError

class ConcreteD(Interface):
    def abstract_method(self, a: int):  # Assumes implicit '-> Any' return type?
        return 123
```
